### PR TITLE
Remove access to underlying contiguous TL buffer from tests

### PR DESCRIPTION
**Pull request by klecki**
Signed-off-by: Krzysztof Lecki <klecki@nvidia.com>

## Description
- [ ] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [x] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [ ] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR
Tests access TL on sample basis instead of using underlying contiguous allocation.
Remove some dead code.

#### Additional information
- Affected modules and functionalities:
Tests

- Key points relevant for the review:

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-2234

### CODE REVIEW

**Comment by klecki** [comment]
!build

File: dali/operators/generic/shapes_test.cc [hunk 1]
```diff
@@ -35,7 +35,7 @@ void GenerateShapeTestInputs(TensorList<Backend> &out, RNG &rng, int num_samples
   }
   out.Reset();
   out.Resize(shape);
-  (void)out.template mutable_data<uint8_t>();
+  out.set_type(TypeInfo::Create<uint8_t>());
```
Comment (banasraf) [inline dali/operators/generic/shapes_test.cc:38]:
```suggestion
  out.set_type(TypeTable::GetTypeInfoFromStatic<uint8_t>());
```

Comment (klecki) [reply dali/operators/generic/shapes_test.cc]: done

**Review by banasraf — APPROVED** [review]

**Comment by klecki** [comment]
!build

**Review by banasraf — APPROVED** [review]

File: dali/operators/bbox/bb_flip_test.cc [hunk 2]
```diff
@@ -100,9 +100,9 @@ const TestSample &FindSample(const TestSample (&dataset)[N], const Roi &roi) {
 template<size_t Idx, typename Backend, size_t N>
 std::unique_ptr<TensorList<Backend>> ToTensorList(const TestSample (&sample)[N]) {
   std::unique_ptr<TensorList<Backend>> tl(new TensorList<Backend>());
-  tl->Resize(uniform_list_shape(N, {kBbStructSize}));
-  auto ptr = tl->template mutable_data<float>();
   for (size_t n = 0; n < N; n++) {
+  tl->Resize(uniform_list_shape(N, {kBbStructSize}));
```
Comment (stiepan) [inline dali/operators/bbox/bb_flip_test.cc:104]: Does this line need to be inside for loop?

Comment (klecki) [reply dali/operators/bbox/bb_flip_test.cc:103]: Moved accidentally, fixed.

**Comment by klecki** [comment]
!build

**Review by stiepan — APPROVED** [review]

**Pull request merged**